File: wlr/SPX.Entities/IEntity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SPX.Core;

namespace SPX.Entities
{
    public interface IEntity
    {
        bool Contains(Point2 target);
        void Draw();
        void Update();
        void LoadContent();
        Point2 GetLocation();
        int EntityType();
        bool IsActive();
        bool IsBlocking();
        void SetInactive();
    }
}

File: proj/src/spx/entities/IEntity.java
package spx.entities;

import spx.bridge.EntityType;
import spx.core.Point2;

public interface IEntity {
    boolean contains(Point2 target);

    void draw();

    void update();

    void loadContent();

    Point2 getLocation();

    EntityType getEntityType();

    boolean isActive();

    boolean isBlocking();

    void setInactive();
}
